providers/utils.go

```diff
@@ -156,14 +157,17 @@ func (i *ProviderInfo) SubstituteWith(vars []paramsbuilder.CatalogVariable) erro
 	return paramsbuilder.NewCatalogSubstitutionRegistry(vars).Apply(i)
 }
 
-func (i *ProviderInfo) GetOption(key string) (string, bool) {
+func (i *ProviderInfo) GetOption(key string) (string, error) {
```

Comment:
This method was not used. I repurposed it and it came in handy.
I verified that server repo doesn't use it either.